contribs/drt/src/main/java/org/matsim/contrib/drt/util/stats/DrtVehicleOccupancyProfiles.java

```diff
@@ -41,7 +43,8 @@
 		//we want the following order on the plot: STAY, RELOCATE, other
 		if (type.equals(DrtStayTask.TYPE)) {
 			return "C";
-		} else if (type.equals(EmptyVehicleRelocator.RELOCATE_VEHICLE_TASK_TYPE)) {
+		} else if (type.equals(EmptyVehicleRelocator.RELOCATE_VEHICLE_TASK_TYPE) ||
+				DrtTaskBaseType.STAY.equals(((DrtTaskType) type).getBaseType().orElse(null))) {
```

Comment:
- What is the purpose of this change? If you need a special sorting for tasks which are available in a specific DRT extension then the special sorting strategy should be provided in that extension (every extension may want something else here).
- There is a shortcut method for line 47: `DrtTaskBaseType.STAY.isBaseTypeOf(type)`
- This outcome order of task types satisfying the `else if` block (lines 46-47) is not deterministic, to get one you need: `return "B" + type.name()`

Reply:
Thanks for your input! I adjusted the code in the latest commit. However, I had to introduce a binding for the profile writers to allow creating instances with other comparators

Reply:
Thanks for addressing my comments!

When it comes to the profile writers, instead of introducing another binding, please just add your own profile writer to mobsim listeners. This way:
- we avoid excessive coupling between different parts of the code
- we can generate as many profiles as we want
Of course, we need to provide a filename to the profile writer so one profile will not overwrite another.